Chapters 1 & 2 Revisions 05-02-2025 (#5113)
dfinity/motoko: doc/md/fundamentals/2-types/14-type-conversions.md

let result4 = intToNat(-5);   // 5

> ```

Reviewer: This file on type conversions could need some simplification and restructuring.

The main point is that Motoko doesn't provide cast operators between types, like other langauges, but for better or worse, just uses functions to do conversions. Then give a couple of examples of to/from and wrapping conversions and a table that shows the nxn matrix of provided conversions between primitive types, with either a name or a just tickmark to indicate availability. Not sure what works best.

For arrays, mention thaw and free (and leave an html comment for use to update that for new base).

For collections in general maybe mention using iterators to convert from one to the other.

I would remove the two lengthy examples converting time to text and text to float. These should be in libraries and if you provided untested code, it will just be a source of cut-n-paste errors.

The time example might make a nice separate example, but not here in the overview of the available conversions.

Maybe just adding simple table of conversion between primitive types would be good and some general

Author: Both examples are tested. The text to float more so that the time example because i freehanded that on icp ninja. The text to float comes from a json library i created that is tested pretty comprehensively with unit tests. I agree they are lengthy but equally valuable because devs will just look it up on the forum and copy and paste from there. Also I think this type conversions page was intended to tackle some of the backlog for missing documentation, might be worth breaking it up

> 

Reviewer: ```suggestion
<!-- new-base specific
```

:::info `Nat` to `Int` conversions functions

> `Int.toNat`, `Int.fromNat`, `Nat.toInt`, `Nat.fromInt` all exist in the base library.

Reviewer 2: ```suggestion
The base library includes functions for certain conversions:`Int.toNat`, `Int.fromNat`, `Nat.toInt`, `Nat.fromInt`.
```

Reviewer: Actually, these functions are only available in the new base library.
Maybe just put the entire admonition in an html comment for now.

:::

> 

Reviewer: ```suggestion
-->
```

Additionally, a method for applying a timezone offset is provided, allowing adjustments such as converting UTC to Eastern Standard Time (EST) by subtracting 5 hours. The final output is formatted as `MM-DD-YYYY HH-MM-SS` for readability.

```motoko
import Time "mo:base/Time";
import Int "mo:base/Int";
import Text "mo:base/Text";

> persistent actor D{

Reviewer 2: actor name should be more descriptive
